src/V3SplitVar.cpp

```diff
@@ -717,7 +717,6 @@ class SplitUnpackedVarVisitor final : public VNVisitor, public SplitVarImpl {
                         adtypep = VN_AS(refp->dtypep()->skipRefp(), UnpackArrayDType);
                     } else {
                         AstSliceSel* const selp = VN_AS(ref.nodep(), SliceSel);
-                        UASSERT_OBJ(selp, ref.nodep(), "Unexpected op is registered");
```

Comment:
Newer gcc warns on this line, so I removed the unnecessary assertion (VN_AS already checked `ref.nodep() == nullptr`).

```
../V3Error.h:567:20: error: ‘this’ pointer is null [-Werror=nonnull]
  567 |     v3errorEndFatal(v3errorBuildMessage( \
      |                    ^
...
../V3SplitVar.cpp:722:25: note: in expansion of macro ‘UASSERT_OBJ’
  722 |                         UASSERT_OBJ(selp, ref.nodep(), "Unexpected op is registered");
      |                         ^~~~~~~~~~~
```